lib/commands/gestures.js

```diff
@@ -2,62 +2,85 @@ import { util } from 'appium/support';
 import { errors } from 'appium/driver';
 
 /**
+ * Set value to the given element.
+ * Note:
+ * This is not exposed as 'macos: setValue' because this is the same as
+ * element.send_keys in W3C WebDriver spec.
  *
- * @param {import('@appium/types').StringRecord} [options={}]
- * @param {string[]} [keyNames]
- * @returns {string|null}
+ * @this {Mac2Driver}
+ * @param {string} elementId Uuid of the element to set value for.
+ * @param {any} [value] Value to set. Could also be an array.
+ * @param {string} [text] Text to set. If both value and text are set then `value` is preferred
+ * @param {number} [keyModifierFlags] If set then the given key modifiers will
+ *                  be applied while the element value is being set. See
+ *                  https://developer.apple.com/documentation/xctest/xcuikeymodifierflags
+ *                  for more details.
  */
-function extractUuid (options = {}, keyNames = ['elementId', 'element']) {
-  for (const name of keyNames) {
-    if (options[name]) {
-      const result = util.unwrapElement(options[name]);
-      if (result) {
-        return result;
-      }
-    }
-  }
-  return null;
-}
+export async function macosSetValue (elementId, value, text, keyModifierFlags) {
+  return await this.wda.proxy.command(`/element/${elementId}/value`, 'POST', {
+    value, text,
+    keyModifierFlags,
+  });
+};
+
 
 /**
- *
- * @param {import('@appium/types').StringRecord} [options={}]
- * @param {string[]} [keyNames]
- * @returns {string|null}
+ * Raise invalid argument error if element id was unset or x and y were unset.
+ * @param {string} [elementId]
+ * @param {number} [x]
+ * @param {number} [y]
+ * @returns {void}
  */
-function requireUuid (options = {}, keyNames = ['elementId', 'element']) {
-  const result = extractUuid(options, keyNames);
-  if (!result) {
-    throw new errors.InvalidArgumentError(`${keyNames[0]} field is mandatory`);
+function requireElementIdOrXY (elementId, x, y) {
+  if (!!elementId || (!!x && !!y)) {
+    return;
   }
-  return result;
+  throw new errors.InvalidArgumentError(`'elementId' or 'x' and 'y' is required.`);
 }
 
 /**
- * Set value to the given element
- *
- * @this {Mac2Driver}
- * @param {import('../types').SetValueOptions} opts
+ * Raise invalid argument error if sourceElementId and destinationElementId were unset
+ * or startX, startY, endX and endY were unset.
+ * @param {string} [sourceElementId]
+ * @param {string} [destinationElementId]
+ * @param {number} [startX]
+ * @param {number} [startY]
+ * @param {number} [endX]
+ * @param {number} [endY]
+ * @returns {void}
  */
-export async function macosSetValue (opts) {
-  const uuid = requireUuid(opts);
-  const { value, text, keyModifierFlags } = opts ?? {};
-  return await this.wda.proxy.command(`/element/${uuid}/value`, 'POST', {
-    value, text,
-    keyModifierFlags,
-  });
-};
+function requireSourceDestWithElementsOrCoordinates (
```

Comment:
I am not sure if these checks are needed on the driver level as they are anyway performed in WDA

Reply:
Well, the mac2 driver level changes the request destination path like:

```
  const url = sourceElementId && destinationElementId
    ? `/wda/element/${sourceElementId}/clickAndDrag`
    : '/wda/clickAndDrag';
```

so either a specific error is made in WDA, but asking `sourceElementId, destinationElementId` or `startX, startY, endX, endY` level (e.g. nothing was provided) should help in the caller (maca2 driver) level to suggest both destination ways.